react/ProductDescription.tsx

```diff
@@ -55,14 +66,31 @@ function ProductDescription(props: PropsWithChildren<Props>) {
   const { handles } = useCssHandles(CSS_HANDLES, { classes: props.classes })
   const intl = useIntl()
   const { product } = useProduct()
+  const { isMobile } = useDevice()
 
   const description = props.description ?? product?.description
 
   if (!description) {
     return null
   }
+  
+  const { collapseContent = true, title, customCollapse = false } = props
+
+  let {desktopLimitCharacters = defaultDesktopCharacters, mobileLimitCharacters = defaultMobileCharacters} = props
+  let limitCharacters = 0;
+  if(customCollapse) {
+    desktopLimitCharacters = Number.isNaN(desktopLimitCharacters) ? defaultDesktopCharacters : desktopLimitCharacters;
+    mobileLimitCharacters = Number.isNaN(mobileLimitCharacters) ? defaultMobileCharacters : mobileLimitCharacters;
+    limitCharacters = isMobile ? mobileLimitCharacters : desktopLimitCharacters;
+  }
 
-  const { collapseContent = true, title } = props
+  const [ filteredDescription, setFilteredDescription ] = useState(description.substring(0, limitCharacters));
```

Comment:
Following the other suggestions I've made, you should make it react to `limitCharacters` and `isCustomCollapsed` (maybe use `useMemo`)